File: 250515/상하로 반복하며 출력/print-in-up-down-reverse.py
N = int(input())

grid = [[0 for _ in range(N)] for _ in range(N)]

for i in range(N):
    if i % 2 == 0:
        for j in range(N):
            grid[j][i] = j + 1

    else:
        for j in range(N - 1, -1, -1):
            grid[j][i] = N - j


for row in grid:
    print("".join(map(str, row)))

File: 251110/상하로 반복하며 출력/print-in-up-down-reverse.js
const fs = require('fs');
const input = fs.readFileSync(0).toString().trim();
const N = Number(input);

for (let i = 0; i < N; i += 1) {
    let row = [];
    for (let j = 0; j < N; j += 1) {
        if (j % 2 === 0) {
            row.push(i + 1);
        } else {
            row.push(N - i);
        }
    }
    console.log(row.join(''));
}
